mlflow/utils/requirements_utils.py

```diff
@@ -409,3 +420,44 @@ def _get_pinned_requirement(package, version=None, module=None):
             version = version_raw
 
     return f"{package}=={version}"
+
+
+class _MismatchedPackageInfo(NamedTuple):
+    package_name: str
+    installed_version: str
+    requirement: str
+
+    def __str__(self):
+        current_status = self.installed_version if self.installed_version else "uninstalled"
+        return f"{self.package_name} (current: {current_status}, required: {self.requirement})"
+
+
+def _check_requirement_satisfied(requirement_str):
+    """
+    If installed package satisfy the requirements, return None,
+    otherwise return an instance of namedtuple `_MismatchPackageInfo`.
+    """
+    _init_modules_and_packages_maps()
+    req = pkg_resources.Requirement.parse(requirement_str)
+    pkg_name = req.name
+
+    try:
+        installed_version = _get_installed_version(pkg_name, _PACKAGES_TO_MODULES.get(pkg_name))
+    except ModuleNotFoundError:
+        return _MismatchedPackageInfo(
+            package_name=pkg_name,
+            installed_version=None,
+            requirement=requirement_str,
+        )
+
+    if len(req.specifier) == 0:
+        return None
+
+    if req.specifier.contains(installed_version):
+        return None
+    else:
+        return _MismatchedPackageInfo(
+            package_name=pkg_name,
+            installed_version=installed_version,
+            requirement=requirement_str,
+        )
```

Comment:
```suggestion
    if req.specifier and !req.specifier.contains(installed_version):
        return _MismatchedPackageInfo(
            package_name=pkg_name,
            installed_version=installed_version,
            requirement=requirement_str,
        )
        
   return None
```

This might be simpler.